pandas/core/indexing.py

```diff
@@ -860,7 +867,7 @@ def _handle_lowerdim_multi_index_axis0(self, tup: Tuple):
         axis = self.axis or 0
         try:
             # fast path for series or for tup devoid of slices
-            return self._get_label(tup, axis=axis)
+            return self._get_label(tup, axis=axis)  # type: ignore
```

Comment:
What error was this throwing? And one below. Sorry if missed in diff

Reply:
axis needs to be int but is type Axis.

probably not yet got the return type for `self.obj._get_axis_number(axis)` L112. This is defined in another module so would rather keep additions to single modules for now.

could add a cast, but that lets you lie to the type checker so can be more dangerous than ignoring for now.

not sure whether mypy will figure this out once more type hints are added.

as long as we keeping using `--warn-unused-ignores` we should know if these become redundant and can be removed. otherwise will probably need to add cast in the future.

Reply:
How hard would it be to just type that return then? Understood in a different module but reason I ask is I think there is buggy axis behavior when using strings that mypy may detect for us if we don't ignore.

To illustrate, these go down different code paths though result is equivalent

```python
In [5]: df = pd.DataFrame(np.ones((2,2)), index=pd.MultiIndex.from_tuples((('a', 'b'), ('c', 'd'))))
In [5]: df.loc[("a", "b")]
Out[8]:
0    1.0
1    1.0
Name: (a, b), dtype: float64

In [9]: df.loc(axis="index")[("a", "b")]
Out[9]:
0    1.0
1    1.0
Name: (a, b), dtype: float64
```

Reply:
> How hard would it be to just type that return then? Understood in a different module but reason I ask is I think there is buggy axis behavior when using strings that mypy may detect for us if we don't ignore.

I accept that we probably have different motivations here.

my motivation is to clean up the io.formats code. there is a lot of code duplication there, and i believe buggy code there as well.

so my view is that to support the refactoring activity, it's better to get, say 80%, of each of the imports to formats typed, than to get 100% of just a few.

so i'd prefer "adding # type: ignore comments to silence errors you don’t want to fix now." https://mypy.readthedocs.io/en/latest/existing_code.html

Reply:
OK sure. I'm just hesitant that `ignore` and `Any` mean we won't ever review in earnest, but sounds like you have a plan here so that doesn't apply

Reply:
> I'm just hesitant that `ignore` and `Any` mean we won't ever review in earnest

I'm sure that in time we will want to increase the strictness of the mypy checks. so hopefully this won't apply.

also, i'm using MonkeyType to annotate with the actual types passed and returned, rather than what we think they should be. This itself is quickly revealing refactoring/cleanup opportunities.

I guess we could categorize bugs into three categories for the purpose of adding type hints.

1. bugs introduced changing code just to appease the type checker. hence i would prefer ignore to cast, and cast to code changes.

2. bugs in the codebase. adding type hints is going to help enormously.

3. bugs that will be added in the future. Again here, i have the opinion that getting, say 80%, of the codebase done quickly is more likely to be a benefit than perfection on the parts that are done.